java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S4784_java.html

```diff
@@ -0,0 +1,90 @@
+with every additional <code>a</code> character added to the input, the time required to evaluate the regex doubles. However, the equivalent regular
+expression, <code>a+</code> (without grouping) is efficiently evaluated in milliseconds and scales linearly with the input size.</p>
+<p>Evaluating user-provided strings as regular expressions opens the door to <a
+href="https://www.owasp.org/index.php/Regular_expression_Denial_of_Service_-_ReDoS">Regular expression Denial of Service (ReDoS)</a> attacks. In the
+context of a web application, attackers can force the web server to spend all of its resources evaluating regular expressions thereby making the
+service inaccessible to genuine users.</p>
+<p> </p>
+<p>Another type of vulnerability can occur when regular expressions are used to validate user input. A regular expression can be used to filter unsafe
+input by either matching a whole input when it is valid (example: the whole string should only contain alphanumeric characters) or by detecting
+dangerous parts of an input. In both cases it is possible to let dangerous values through. For example, searching for <code>&lt;script&gt;</code> tags
+in some HTML code with the regular expression <code>.*&lt;script&gt;.*</code> will miss <code>&lt;script id="test"&gt;</code>.</p>
+<p> </p>
+<p>This rule flags any regular expression execution or compilation for review.</p>
+<h2>Ask Yourself Whether</h2>
+<ul>
+  <li> a user input string is executed as a regular-expression, or it is inserted in a regular expression. </li>
+  <li> a user can provide a string which will be analyzed by a regular expression. </li>
+  <li> your regular expression engine performance decrease with specially crafted inputs and regular expressions. </li>
+  <li> the regular expression is used to validate unsafe input, but it does not detect all dangerous values. </li>
+</ul>
+<p>You may be at risk if you answered yes to any of those questions.</p>
+<h2>Recommended Secure Coding Practices</h2>
+<p>Avoid executing a user input string as a regular expression. If this is required, restrict the allowed regular expressions.</p>
+<p>Check whether your regular expression engine (the algorithm executing your regular expression) has any known vulnerabilities. Search for
+vulnerability reports mentioning the one engine you're are using.</p>
+<p>Test your regular expressions with techniques such as equivalence partitioning, and boundary value analysis, and test for robustness. Try not to
+make complex regular expressions as they are difficult to understand and test. Note that some regular expression engines will match only part of the
+input if no anchors are used. In PHP for example <code>preg_match("/[A-Za-z0-9]+/", $text)</code> will accept any string containing at least one
+alphanumeric character because it has no anchors.</p>
+<h2>Questionable Code Example</h2>
+<pre>
+  import java.util.regex.Pattern;
```

Comment:
I really don't like this formatting. It mixes imports and content which should be in the body of a method. I think we should improve it and split the two examples. The second one preceded by one sentence explicitly mentioning the bean validation context, without having it being a simple comment before new imports.